lib/eslint/eslint.js

```diff
@@ -314,7 +358,33 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath) && hasUnstableTSConfigFlag;
+
+    const isBun = isRunningInBun();
+
+    const isDeno = isRunningInDeno();
+
+    if (isTS && !isDeno && !isBun) {
+
+        const createJiti = await import("jiti").then(jitiModule => jitiModule?.createJiti ?? jitiModule.default, () => {
+            throw new Error("The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.");
+        });
+
+        /*
+         * Disabling `moduleCache` allows us to reload a
+         * config file when the last modified timestamp changes.
+         */
+
+        const jiti = createJiti(__filename, { moduleCache: false });
+
+        const config = /** @type {{ default?: FlatConfigArray }} */ (await jiti.import(fileURL.href));
```

Comment:
`FlatConfigArray` is a class used internally by ESLint, not the the type of a config file, for which we don't have a typing yet (it could be a single object or an array). Also, apparently `jiti.import` will not always set a `default` property when importing a (CommonJS?) module.